lib/swoosh/attachment.ex

```diff
@@ -3,9 +3,16 @@ defmodule Swoosh.Attachment do
   Struct representing an attachment in an email.
   """
 
-  defstruct filename: nil, content_type: nil, path: nil, type: nil, headers: [], data: nil
+  defstruct [:filename, :content_type, :path, :type, :headers, :data]
 
-  @type t :: %__MODULE__{}
+  @type t :: %__MODULE__{
+    filename: String.t,
+    content_type: String.t,
+    path: String.t,
+    type: :inline | :attachment,
+    headers: [{String.t, String.t}],
+    data: binary
```

Comment:
according to `get_content/2` data should be `binary | nil`

Reply:
Then `path` should be allowed to be `nil` too, right? When it's in-memory data.

Reply:
Thanks for the review BTW!

Reply:
good catch.

That brings me another little optimization, since data and path cannot be nil at the same time.
You can add a function clause like:

`def get_content(%__MODULE__{data: nil, path: nil})`

that spits out an exception or {:error, term}, or anything like, just to be sure user is alerted by your layer (otherwise a File.read! exception will raise)

Reply:
Right. I'll add a new error for that. Thanks @xadhoom .